### ISSUE TITLE
In a List, also show the Empty component when no results and only a permanent filter

### ISSUE THREAD

**Issue by christiaanwesterbeek** [marmelab/react-admin#8320]
**Is your feature request related to a problem? Please describe.**

The `filter` prop for List is considered a permanent filter. When an empty={<Empty ?} prop is supplied to the List with such a permanent filter (and no results), the Empty component will not render. The documentation about it [says](https://marmelab.com/react-admin/List.html#empty-empty-page-component) (emphasis mine).

> When there is no result, **and there is no active filter**, ..., react-admin displays ...

**Describe the solution you'd like**

Have you considered or will you consider showing the Empty page in this case? The permanent filter is not a user decision, so he should be invited to create a record within the fixed scope of the data.

**Describe alternatives you've considered**

Alternatively, I could add something underneath the Datagrid that fetches the list context and decide for it's own to show the Empty component...

**Additional context**

-

**Comment by slax57** [comment]
I agree, this would be a welcome improvement.

Btw, I'm not 100% sure but I believe `Datagrid` has his own `empty` element, and there it is displayed from the moment there is no result, filters or not.

So basically we should do the same thing for all `List` child, and not only `Datagrid`.

**Comment by christiaanwesterbeek** [comment]
You're right. Problem solved by passing my `<Empty />` element tot the `empty` prop of the `Datagrid` instead.

**Comment by slax57** [comment]
Thanks @christiaanwesterbeek for this feedback.
Still, I don't really see a good reason why the `List`'s `empty` element would not be displayed when there are active filters.
Hence I'll keep this feature request open, if anyone wants to have a look.

**Comment by slax57** [comment]
Coming back to this issue (as I'm doing a bit of triaging/cleaning), I realize we currently don't expose the permanent filters (i.e. the `filter` prop) in the `ListContext`, so currently `<ListView>` can't tell if the active filters are permanent filters or not.

However it shouldn't be too hard to forward `filter` from `<List>` to `<ListView>` to make that comparison possible.
If anyone wants to give that a try a PR would be welcome.

**Comment by fury0928** [comment]
@slax57 can I take on this issue?

**Comment by djhi** [comment]
@eureka928 Thanks for asking :) We don't assign issues so you can work on anything you like.

**Comment by slax57** [comment]
Unless I'm mistaken, I believe https://github.com/marmelab/react-admin/pull/11124 attempted to address this exact same issue. And I made another test today ([stackblitz](https://stackblitz.com/edit/github-ik9btczd?file=src%2Fposts%2FPostList.tsx), setting a permanent filter on the Post List (`{ commentable: false }`)), which made me realize this bug might already have been fixed!

@eureka928 @christiaanwesterbeek can you confirm?

**Comment by Bhavanapulla** [comment]
Hi! I'd like to work on this issue
My understanding - 
- Currently, when a List has permanent filters and returns no results, nothing is displayed
- Expected: The Empty component should be shown in this case, similar  to when there are no filters
I'll investigate the List component logic and submit a PR 
Please let me know if: My understanding is correct . Anyone is already working on it
Thanks!

**Comment by slax57** [comment]
As I mentioned above, to me, the issue has already been fixed. 
@Bhavanapulla  You can help by first trying to reproduce the issue, and sharing your reproduction if you manage to do it. If you can't reproduce, then please also share this result and we'll close this issue. Thanks.
